lib/install/management/configure.go

```diff
@@ -258,21 +258,25 @@ func (d *Dispatcher) deleteSnapshotByRef(snapshot *types.ManagedObjectReference,
 	return nil
 }
 
-// tryCreateSnapshot try to create upgrade snapshot. It will check if upgrade snapshot already exists. If exists, return error.
-// if succeed, return snapshot refID
-func (d *Dispatcher) tryCreateSnapshot(name, desc string) (*types.ManagedObjectReference, error) {
+// createSnapshot will create a snapshot of the VCH Appliance
+func (d *Dispatcher) createSnapshot(name, desc string) (*types.ManagedObjectReference, error) {
 	defer trace.End(trace.Begin(name, d.op))
 
 	// TODO detect whether another upgrade is in progress & bail if it is.
 	// Use solution from https://github.com/vmware/vic/issues/4069 to do this either as part of 4069 or once it's closed
 
 	info, err := d.appliance.WaitForResult(d.op, func(ctx context.Context) (tasks.Task, error) {
-		return d.appliance.CreateSnapshot(ctx, name, desc, true, false)
+		return d.appliance.CreateSnapshot(ctx, name, desc, false, false)
 	})
 	if err != nil {
 		return nil, errors.Errorf("Failed to create upgrade snapshot %q: %s.", name, err)
 	}
-	return info.Entity, nil
+	snap, ok := info.Result.(types.ManagedObjectReference)
+	if !ok {
+		// TODO: what to do in this situation..
```

Comment:
What should we do here? Also, is there a difference b/w `info.Entity` and `info.Result`?

Reply:
> Also, is there a difference b/w info.Entity and info.Result?

Oh I see, I think it's because https://github.com/vmware/vic/pull/7574#issuecomment-375498998

Reply:
`info.Entity` is the object acted on (vm in this case) -- `info.Result` is what resulted..so in this case the snapshot --- my _assumption_ is that if this would not work....it would have failed earlier during the task...

Reply:
So to put it another way...if there was no valid result then we wouldn't even make it here...

Reply:
Agreed. Perhaps the type check isn't needed, because the `info.Result` should be a moref if `CreateSnapshot` returned no error.

Reply:
it's a cast not a check -- it returns as `AnyType` and we have to cast to the a moref -- if this fails we should return an error so that we fail the upgrade...again I doubt we'd ever hit it..but as soon as I say that...we'll hit it...so I'll push an update in a minute

Reply:
Haha, thanks for explaining!